build/root/Makefile.generated_files

```diff
@@ -805,3 +701,173 @@ sinclude $(META_DIR)/$(CONVERSION_GEN).mk
+.PHONY: $(META_DIR)/$(OPENAPI_GEN).mk
+$(META_DIR)/$(OPENAPI_GEN).mk:
+	mkdir -p $(@D);                                                       \
+	(echo -n "$(OPENAPI_GEN): ";                                          \
+	 ./hack/run-in-gopath.sh go list                                      \
+	     -f '{{.ImportPath}}{{"\n"}}{{range .Deps}}{{.}}{{"\n"}}{{end}}'  \
+	     ./vendor/k8s.io/code-generator/cmd/openapi-gen                   \
+	     | grep --color=never "^$(PRJ_SRC_PATH)/"                         \
+	     | xargs ./hack/run-in-gopath.sh go list                          \
+	         -f '{{$$d := .Dir}}{{$$d}}{{"\n"}}{{range .GoFiles}}{{$$d}}/{{.}}{{"\n"}}{{end}}'  \
+	     | paste -sd' ' -                                                 \
+	     | sed 's/ / \\=,/g'                                              \
+	     | tr '=,' '\n\t'                                                 \
+	     | sed "s|$$(pwd -P)/||";                                         \
+	) > $@.tmp;                                                           \
+	if ! cmp -s $@.tmp $@; then                                           \
+	    if [[ "$(DBG_CODEGEN)" == 1 ]]; then                              \
+	        echo "DBG: $(OPENAPI_GEN).mk changed";                        \
+	    fi;                                                               \
+	    cat $@.tmp > $@;                                                  \
+	    rm -f $@.tmp;                                                     \
+	fi
+
+# Include dependency info for the generator tool.  This will cause the rule of
+# the same name to be considered and if it is updated, make will restart.
+sinclude $(META_DIR)/$(OPENAPI_GEN).mk
+
+# How to build the generator tool.  The deps for this are defined in
+# the $(OPENAPI_GEN).mk, above.
+#
+# A word on the need to touch: This rule might trigger if, for example, a
+# non-Go file was added or deleted from a directory on which this depends.
+# This target needs to be reconsidered, but Go realizes it doesn't actually
+# have to be rebuilt.  In that case, make will forever see the dependency as
+# newer than the binary, and try to rebuild it over and over.  So we touch it,
+# and make is happy.
+$(OPENAPI_GEN):
+	hack/make-rules/build.sh ./vendor/k8s.io/code-generator/cmd/openapi-gen
+	touch $@
+
+#
+# bindata generation
+#
+
+# The result files.
+# TODO: Automatically find these and regen IFF needed.
+BINDATA_OUTFILES := test/e2e/generated/bindata.go \
+                    pkg/generated/bindata.go
+
+# The tool used to generate bindata files.
+BINDATA_GEN := $(BIN_DIR)/go-bindata
+
+# A wrapper script that generates all bindata files.  It is fast enough that we
+# don't are.
+BINDATA_SCRIPT := hack/generate-bindata.sh
+
+# This rule is the user-friendly entrypoint for bindata generation.
+.PHONY: gen_bindata
+gen_bindata: $(BINDATA_OUTFILES)
+
+$(BINDATA_OUTFILES): $(BINDATA_GEN) $(BINDATA_SCRIPT)
```

Comment:
I don't understand how `make` decides when to run this rule, but I seem to be able to confuse it.

I checked out your PR locally, then ran a few commands:
```console
$ git status
On branch pr/62151
nothing to commit, working tree clean
$ make clean
+++ [0405 15:31:25] Verifying Prerequisites....          
+++ [0405 15:31:26] Removing _output directory
$ make generated_files
+++ [0405 15:31:51] Building go targets for linux/amd64: 
    ./vendor/k8s.io/code-generator/cmd/deepcopy-gen                                                                                                                                                                 
+++ [0405 15:31:57] Building go targets for linux/amd64:
    ./vendor/k8s.io/code-generator/cmd/defaulter-gen
+++ [0405 15:32:01] Building go targets for linux/amd64:
    ./vendor/k8s.io/code-generator/cmd/conversion-gen
+++ [0405 15:32:07] Building go targets for linux/amd64:
    ./vendor/k8s.io/code-generator/cmd/openapi-gen
+++ [0405 15:32:12] Building go targets for linux/amd64:
    ./vendor/github.com/jteeuwen/go-bindata/go-bindata
$ git status
On branch pr/62151                                       
nothing to commit, working tree clean
$ vim translations/kubectl/en_US/LC_MESSAGES/k8s.po
# edit some lines
$ git status
On branch pr/62151
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   translations/kubectl/en_US/LC_MESSAGES/k8s.po

no changes added to commit (use "git add" and/or "git commit -a")
$ git diff
diff --git a/translations/kubectl/en_US/LC_MESSAGES/k8s.po b/translations/kubectl/en_US/LC_MESSAGES/k8s.po
index 4c2f11466b..5d9a2cdf00 100644
--- a/translations/kubectl/en_US/LC_MESSAGES/k8s.po
+++ b/translations/kubectl/en_US/LC_MESSAGES/k8s.po
@@ -22,9 +22,9 @@ msgstr ""
 #: pkg/kubectl/cmd/create_clusterrolebinding.go:35
 msgid ""
 "\n"
-"\t\t  # Create a ClusterRoleBinding for user1, user2, and group1 using the "
+"\t\t  # Creat a ClusterRoleBinding for user1, user2, and group1 using the "
 "cluster-admin ClusterRole\n"
-"\t\t  kubectl create clusterrolebinding cluster-admin --clusterrole=cluster-"
+"\t\t  kubectl creat clusterrolebinding cluster-admin --clusterrole=cluster-"
 "admin --user=user1 --user=user2 --group=group1"
 msgstr ""
 "\n"
$ make generated_files
$ git diff
diff --git a/pkg/generated/bindata.go b/pkg/generated/bindata.go
index 7cbfc6154f..a657a24ae7 100644
--- a/pkg/generated/bindata.go
+++ b/pkg/generated/bindata.go
@@ -6975,9 +6975,9 @@ msgstr ""
 #: pkg/kubectl/cmd/create_clusterrolebinding.go:35
 msgid ""
 "\n"
-"\t\t  # Create a ClusterRoleBinding for user1, user2, and group1 using the "
+"\t\t  # Creat a ClusterRoleBinding for user1, user2, and group1 using the "
 "cluster-admin ClusterRole\n"
-"\t\t  kubectl create clusterrolebinding cluster-admin --clusterrole=cluster-"
+"\t\t  kubectl creat clusterrolebinding cluster-admin --clusterrole=cluster-"
 "admin --user=user1 --user=user2 --group=group1"
 msgstr ""
 "\n"
diff --git a/translations/kubectl/en_US/LC_MESSAGES/k8s.po b/translations/kubectl/en_US/LC_MESSAGES/k8s.po
index 4c2f11466b..5d9a2cdf00 100644
--- a/translations/kubectl/en_US/LC_MESSAGES/k8s.po      
+++ b/translations/kubectl/en_US/LC_MESSAGES/k8s.po      
@@ -22,9 +22,9 @@ msgstr ""                              
 #: pkg/kubectl/cmd/create_clusterrolebinding.go:35      
 msgid ""                                                
 "\n"                                                    
-"\t\t  # Create a ClusterRoleBinding for user1, user2, and group1 using the "
+"\t\t  # Creat a ClusterRoleBinding for user1, user2, and group1 using the "
 "cluster-admin ClusterRole\n"                           
-"\t\t  kubectl create clusterrolebinding cluster-admin --clusterrole=cluster-"
+"\t\t  kubectl creat clusterrolebinding cluster-admin --clusterrole=cluster-"
 "admin --user=user1 --user=user2 --group=group1"        
 msgstr ""                                               
 "\n"
$ git checkout HEAD translations/kubectl/en_US/LC_MESSAGES/k8s.po
$ make generated_files
$ git diff
diff --git a/pkg/generated/bindata.go b/pkg/generated/bindata.go
index 7cbfc6154f..a657a24ae7 100644                      
--- a/pkg/generated/bindata.go                           
+++ b/pkg/generated/bindata.go                           
@@ -6975,9 +6975,9 @@ msgstr ""                          
 #: pkg/kubectl/cmd/create_clusterrolebinding.go:35      
 msgid ""                                                
 "\n"                                                    
-"\t\t  # Create a ClusterRoleBinding for user1, user2, and group1 using the "
+"\t\t  # Creat a ClusterRoleBinding for user1, user2, and group1 using the "
 "cluster-admin ClusterRole\n"                           
-"\t\t  kubectl create clusterrolebinding cluster-admin --clusterrole=cluster-"
+"\t\t  kubectl creat clusterrolebinding cluster-admin --clusterrole=cluster-"
 "admin --user=user1 --user=user2 --group=group1"        
 msgstr ""                                               
 "\n"
```

So basically, if I update one of the input files, then make the generated files, and then revert my changes to the input file, it doesn't regenerate the output again.

Reply:
ah, yeah.  I tried taking a shortcut, but not too short of a shortcut.  Instead I will take the real shortcut.  fixed in next push.